### DISCUSSION TITLE
Cache OverlayIndex Update

### DISCUSSION THREAD

**Discussion by K4rn4ge** [runelite/runelite#17268] [Ideas]
Hello.

I am working on a plugin that uses OverlayIndex to modify cache files, but I am running into two issues.

Issue one:

I can override index 7 archives fine, as they all contain one file. However, if I want to override an NPC file in Index 2 Archive 9, I can't specify what fileId to override.

Issue two:

With issue one in the way, I figured I'd just repack the entire archive with my changes to see if that would work, but the loadData method checks the hash of the archiveId AND fileId, which since we can't specify files, only indexes and archives, fails the hash comparison check.

So basically, I'm wondering if we can get OverlayIndex updated to use fileId as well.

**Comment by Adam-** [comment]
You can't use OverlayIndex from a plugin because it has no API. This is an intentional limitation currently, adding an API for it would let plugins make a bunch of custom content which we are supposed to avoid. It is currently just used to replace some scripts, which doesn't require file replacement, and so it does not do that. I don't see a reason to expand it to include files right now.

**Reply by K4rn4ge** [reply]
That makes sense. I was just looking to add the older NPC models and animations from 2006 as I saw some requests for it on Reddit. May have expanded to other data from different revisions if the community wanted it.

The plugin part of this would've just been setting NPC animations to the old ones (server sent animations like attacking/defending/death/etc) as the rest would've been done with cache editing/overlaying.

Do you think you could add it if we got the all clear from Jagex and/or if there was enough community support? If it does get approved we could add some sort of "verified" file check which would only allow approved files to go through the override process? Just an idea.

Either way thanks for the reply @Adam-

**Reply by K4rn4ge** [reply]
Assuming you'd want to make the change: Would it be possible just for me to get access to the changed class that way no one else could load anything inappropriate? Maybe a private repo or fork? I'd only need the change long enough to get the data fully loaded and take pics/gifs to show that it's possible.

If we can't do the above or the suggestion from my previous reply, could you let me know what would be needed to potentially make this happen? I'd like to avoid having to heavily modify a 2006 client if at all possible.
